Extensions/quick_tags.js

```diff
@@ -300,34 +300,109 @@ XKit.extensions.quick_tags = new Object({
+						"</div>" +
+					"</div>";
 
 		$("#xkit-quick-tags-window").remove();
+		
+		// FIXME: Should these from the end of the function be added up here too..?¿?
+		// XKit.extensions.quick_tags.user_on_box = true;
+		// XKit.extensions.quick_tags.current_button = $(obj);
+		//			^
+		// With this one copied up here, hovering over the quick-tags icon
+		// gets a spinning "loading" animation rather than nothing, so.. progress ??
+		XKit.extensions.quick_tags.current_button = $(obj);
+		
+		// Should we copy the existing tags into the text box?
+		if(XKit.extensions.quick_tags.preferences.append_not_edit.value !== true) {
+			
+			// If so...
+			// Find the post object.
+			XKit.interface.switch_control_button($(XKit.extensions.quick_tags.current_button), true);
+			// XKit.window.show("Oops", "Test point ONE<br/>", "error", "<div id=\"xkit-close-message\" class=\"xkit-button default\">OK</div>");
+			
+			var m_button = Xkit.extensions.quick_tags.current_button;
+			// FIXME: This test point window never appears. Why does initializing m_button freeze up the extension?
+			XKit.window.show("Oops", "Test point TWO<br/>", "error", "<div id=\"xkit-close-message\" class=\"xkit-button default\">OK</div>");
+			
+			var m_post = XKit.interface.find_post($(m_button).attr('data-post-id'));
+			// XKit.window.show("Oops", "Test point THREE<br/>", "error", "<div id=\"xkit-close-message\" class=\"xkit-button default\">OK</div>");
+			
+			// Fetch info about it!
+			if (!m_post.error) {
+				// XKit.window.show("Oops", "Test point FOUR<br/>", "error", "<div id=\"xkit-close-message\" class=\"xkit-button default\">OK</div>");
+				
+				XKit.interface.fetch(m_post, function(data) {
+					
+					// Use Interface to get the post's tags:
+					var m_tags = data.data.post.tags;
+					// XKit.window.show("Oops", "Test point FIVE<br/>", "error", "<div id=\"xkit-close-message\" class=\"xkit-button default\">OK</div>");
+					
+					// Format the tags:
+					if (m_tags === "undefined" || typeof m_tags === "undefined" || m_tags == "null") {
+						m_tags = "";
+					}
+					
+					if (m_tags.indexOf(",") != -1) {
+						
+						var split_tags = m_tags.split(",");
+						var new_data = "";
+						
+						for (var i=0;i<split_tags.length;i++) {
+							
+							if (split_tags[i]) {
+								if (new_data === "") {
+									new_data = split_tags[i];
+								} else {
+									new_data = new_data + "," + split_tags[i];
+								}
+							}
+							
+						}
+						// XKit.window.show("Oops", "Test point SIX<br/>", "error", "<div id=\"xkit-close-message\" class=\"xkit-button default\">OK</div>");
+						
+						m_tags = new_data;
+						
+					}
+					// XKit.window.show("Oops", "Test point SEVEN<br/>", "error", "<div id=\"xkit-close-message\" class=\"xkit-button default\">OK</div>");
+					
+					// Done formatting, place existing tags in the quicktags pop-up's text box:
+					m_html = m_html.replace("separated tags\" ", "separated tags\" value=\"" + m_tags + "\" ");
```

Comment:
Since you modify m_html up here asynchronously, lines 404 at the least needs to be moved up here so that the pop-up with the current tags is displayed.